subprojects/core/src/main/java/org/gradle/configuration/project/LifecycleProjectEvaluator.java

```diff
@@ -44,9 +43,9 @@ public void evaluate(final ProjectInternal project, final ProjectStateInternal s
             return;
         }
 
-        projectConfigurator.projectBuildOperationAction(project, new Action<BuildOperationContext>() {
+        projectConfigurator.projectBuildOperation(new ProjectConfigurator.ConfigureProjectBuildOperation(project) {
```

Comment:
As discussed, I will address it as part of https://github.com/gradle/gradle/issues/1821